[Doc] Convert configuring-autoscaling.ipynb back to markdown docs

EagleLo opened the pull request:
<!-- Thank you for your contribution! Please review https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR. If you don't have the access to it, we will shortly find a reviewer and assign them to your PR. -->

## Why are these changes needed?

This PR converts the configuring-autoscaling.ipynb notebook back to markdown documentation as part of the effort to remove doctests from documentation and improve maintainability.

**Changes made:**
- Converted `configuring-autoscaling.ipynb` to `configuring-autoscaling.md`
- Restored the previous markdown version from git history and updated it for KubeRay v1.4.0
- Updated worker pod naming format to match KubeRay v1.4.0: `raycluster-autoscaler-small-group-worker-xxxxx`
- Added `autoscalerOptions.version: v2` configuration for KubeRay >= 1.4.0
- Updated Autoscaler V2 prerequisites to mention KubeRay v1.4.0
- Added `kind delete cluster` to cleanup steps
- Removed the original notebook file after conversion
- All quickstart steps have been tested and verified to work with KubeRay v1.4.0 operator

## Related issue number

Closes #54077

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [x] I've added any new APIs to the API Reference. For example, if I added a
           method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/

**Manual Testing:**
✅ All quickstart steps (1-8) have been manually tested and verified to work

Step 1: Create a Kubernetes cluster with Kind
<img width="678" alt="Screenshot 2025-06-25 at 2 39 35 PM" src="https://github.com/user-attachments/assets/06256c3d-0250-404b-97ef-d034b3c01cd8" />

Step 2: Install the KubeRay operator
<img width="829" alt="Screenshot 2025-06-25 at 2 41 02 PM" src="https://github.com/user-attachments/assets/3b3e00e4-943b-4db9-af5b-482b02210c28" />

Step 3: Create a RayCluster custom resource with autoscaling enabled
<img width="815" alt="Screenshot 2025-06-25 at 2 41 40 PM" src="https://github.com/user-attachments/assets/1572c66e-0f5f-4bf5-b65a-4a75e65d8bdc" />

Step 4: Verify the Kubernetes cluster status
<img width="828" alt="Screenshot 2025-06-25 at 2 47 50 PM" src="https://github.com/user-attachments/assets/51f5ae53-edbe-400f-a503-b97d97431764" />

Step 5: Trigger RayCluster scale-up by creating detached actors
kubectl exec -it $HEAD_POD -- python3 /home/ray/samples/detached_actor.py actor1
kubectl exec -it $HEAD_POD -- python3 /home/ray/samples/detached_actor.py actor2
<img width="833" alt="Screenshot 2025-06-25 at 3 02 55 PM" src="https://github.com/user-attachments/assets/e7832fe0-a5fa-4ba5-b4e5-05e5b34180cd" />
<img width="825" alt="Screenshot 2025-06-25 at 3 03 08 PM" src="https://github.com/user-attachments/assets/8ef0463b-c521-48cd-b9c8-dc74ee2534f7" />
<img width="881" alt="Screenshot 2025-06-25 at 3 07 15 PM" src="https://github.com/user-attachments/assets/6c757f2f-297c-4730-932e-32534707ad38" />

Step 6: Trigger RayCluster scale-down by terminating detached actors
<img width="887" alt="Screenshot 2025-06-25 at 3 08 47 PM" src="https://github.com/user-attachments/assets/ec45e423-afa5-48c0-961d-f8312639998d" />
<img width="887" alt="Screenshot 2025-06-25 at 3 10 50 PM" src="https://github.com/user-attachments/assets/934b2859-67e0-49a7-acde-9ea549642eaf" />

Step 7: Ray Autoscaler observability
<img width="889" alt="Screenshot 2025-06-25 at 3 13 14 PM" src="https://github.com/user-attachments/assets/7a93b4d6-bde7-4cfa-a596-0577418cca95" />

Step 8: Clean up
<img width="875" alt="Screenshot 2025-06-25 at 3 14 38 PM" src="https://github.com/user-attachments/assets/88b18cb6-a047-4d34-b5fe-2db555381f94" />

<img width="892" alt="Screenshot 2025-06-25 at 3 14 45 PM" src="https://github.com/user-attachments/assets/2e52fc90-a7e3-46ad-a5dd-72e5f8260348" />

EagleLo committed:
[Doc][KubeRay] Convert configuring-autoscaling.ipynb back to markdown docs

Signed-off-by: Eagle Lo <eagle.lo@samsara.com>

EagleLo committed:
[Doc][KubeRay] Update Autoscaler V2 prerequisites to v1.4.0

Signed-off-by: Eagle Lo <eagle.lo@samsara.com>

EagleLo committed:
[Doc][KubeRay] Remove configuring-autoscaling.ipynb after converting to markdown

Signed-off-by: Eagle Lo <eagle.lo@samsara.com>

kevin85421 reviewed (commented):
> Restored the previous markdown version from git history and updated it for KubeRay v1.4.0

This may miss some commits. You can check the commit history here:
https://github.com/ray-project/ray/commits/master/doc/source/cluster/kubernetes/user-guides/configuring-autoscaling.ipynb

EagleLo committed:
[Doc][KubeRay] Incorporate missing commits from June updates

Signed-off-by: Eagle Lo <eagle.lo@samsara.com>

MortalHappiness reviewed (commented):
Please rerun all steps in the doc with KubeRay v1.4.0 and correct any wrong information in the doc. Especially check for the resource names. Ping me again when you are done. Thanks!

EagleLo committed:
[Doc][KubeRay] Fix resource naming and update to v1.4.0 URLs

Signed-off-by: Eagle Lo <eagle.lo@samsara.com>

EagleLo commented:
@MortalHappiness  Hi! I've addressed your feedback and pushed the updates: 1. Fixed resource naming, 2. Updated to v1.4.0 URL, 3.Updated all references. 
However, I discovered an issue during testing:
When testing with KubeRay v1.4.0 operator + v1.4.0 YAML, the Ray head container crashes with:
`ValueError: Attempting to cap object store memory usage at 1564262 bytes, but the minimum allowed is 78643200 bytes.`
This appears to be a memory constraint issue in Kind environments. The v1.4.0 YAML has tighter memory limits that cause Ray to fail in resource-constrained environments.
Can you help take a look at this?

step 1-3:
<img width="885" alt="step 1-3" src="https://github.com/user-attachments/assets/e0a3b10c-c88a-4b64-9a94-34014f439fd2" />

step 4:
<img width="628" alt="step 4" src="https://github.com/user-attachments/assets/fa6489e4-7c83-41e2-8b7d-58bf75040711" />

MortalHappiness commented on doc/source/cluster/kubernetes/user-guides/configuring-autoscaling.md:
````diff
@@ -0,0 +1,495 @@
+```
+
+The RayCluster has one head Pod and zero worker Pods. The head Pod has two containers: a Ray head container and a Ray Autoscaler sidecar container.
+Additionally, the [ray-cluster.autoscaler.yaml](https://github.com/ray-project/kuberay/blob/v1.3.0/ray-operator/config/samples/ray-cluster.autoscaler.yaml) includes a ConfigMap named `ray-example` that houses two Python scripts: `detached_actor.py` and `terminate_detached_actor.py`.
````
```suggestion
Additionally, the [ray-cluster.autoscaler.yaml](https://github.com/ray-project/kuberay/blob/v1.4.0/ray-operator/config/samples/ray-cluster.autoscaler.yaml) includes a ConfigMap named `ray-example` that houses two Python scripts: `detached_actor.py` and `terminate_detached_actor.py`.
```

Please reexampine the whole doc to replace all v1.3.0 string to v1.4.0. Thanks.

kevin85421 committed:
Apply suggestions from code review

Signed-off-by: Kai-Hsun Chen <kaihsun@apache.org>

kevin85421 commented:
cc @dayshah would you mind approving this PR? Thanks.
